web/src/main/java/com/epam/esm/controller/UserController.java

```diff
@@ -77,11 +90,49 @@ public UserController(UserService userService,
      * @return JSON {@link ResponseEntity} with {@link HateoasModel} object that contains {@link UserDto} object
      */
     @GetMapping("/{id}")
+    @PreAuthorize("hasRole('ADMIN')")
     public ResponseEntity<HateoasModel<UserDto>> getUserById(@PathVariable("id") Long id) {
         UserDto userByIdService = userService.findUserByIdService(id);
         HateoasModel<UserDto> model = new HateoasModel<>(userByIdService);
         HateoasModel<UserDto> build = model.build(modelHateoasProvider, userByIdService, 1L);
 
         return new ResponseEntity<>(build, HttpStatus.OK);
     }
+
+    /**
+     * Create a new user.
+     * Access is allowed to everyone.
+     *
+     * @param userDto {@link UserDto} instance
+     * @throws InvalidEntityException in case when passed DTO object contains invalid data
+     * @throws EntityAlreadyExistsException in case when user with specified username already exists
+     * @return JSON {@link ResponseEntity} with {@link HateoasModel} object that contains {@link UserDto} object
+     */
+    @PostMapping("/signup")
```

Comment:
I think that UserController shouldn't have Authorization responsibility. Let's extract this logic to new, AuthenticationController

Reply:
fixed